### ISSUE TITLE
Bug report: Dropbox errors when in user directory containing dropbox folder

### ISSUE THREAD

**Issue by wickles** [caskroom/homebrew-cask#25496]
Cask `dropbox` fails to show correct output for `info` and `install`, when in user directory containing subfolder `Dropbox`. Works fine when switching to another folder not containing subfolder `Dropbox`, e.g. `/tmp/`. 

<details><summary>Output of `brew cask <command> --verbose`</summary>

```
> % brew cask install dropbox --verbose
Error: File '/Users/alex/dropbox' is not a plain file
Error: Kernel.exit

> % brew cask info dropbox --verbose
Error: File '/Users/alex/dropbox' is not a plain file
Error: Kernel.exit

> % cd /tmp
> % brew cask info dropbox
dropbox: latest
https://www.dropbox.com/
/opt/homebrew-cask/Caskroom/dropbox/latest (68B)
From: https://github.com/caskroom/homebrew-cask/blob/master/Casks/dropbox.rb
==> Name
Dropbox
==> Artifacts
Dropbox.app (app)
```
</details>

<details><summary>Output of `brew doctor`</summary>

```
> % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: Anaconda is known to frequently break Homebrew builds, including Vim and
MacVim, due to bundling many duplicates of system and Homebrew-available
tools.

If you encounter a build failure please temporarily remove Anaconda
from your $PATH and attempt the build again prior to reporting the
failure to us. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
  /Users/alex/anaconda/bin/curl-config
  /Users/alex/anaconda/bin/freetype-config
  /Users/alex/anaconda/bin/libdynd-config
  /Users/alex/anaconda/bin/libpng-config
  /Users/alex/anaconda/bin/libpng16-config
  /Users/alex/anaconda/bin/python-config
  /Users/alex/anaconda/bin/python2-config
  /Users/alex/anaconda/bin/python2.7-config
  /Users/alex/anaconda/bin/xml2-config
  /Users/alex/anaconda/bin/xslt-config
  /usr/local/pspdev/bin/psp-config

Warning: Python is installed at /Library/Frameworks/Python.framework

Homebrew only supports building against the System-provided Python or a
brewed Python. In particular, Pythons installed to /Library can interfere
with other software installs.

Warning: You have MacPorts or Fink installed:
  /opt/local/bin/port

This can cause trouble. You don't have to uninstall them, but you may want to
temporarily move them out of the way, e.g.

  sudo mv /opt/local ~/macports

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python

Warning: Homebrew's sbin was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting the PATH for example like so
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc

Warning: Some installed formula are missing dependencies.
You should `brew install` the missing dependencies:
  brew install shared-mime-info utf8proc

Run `brew missing` for more details.
```
</details>

<details><summary>Output of `brew cask doctor`</summary>

```
> % brew cask doctor
==> macOS Release:
10.12
==> Hardware Architecture:
intel-64
==> Ruby Version:
2.0.0-p648
==> Ruby Path:
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
==> Homebrew Version:
Homebrew 1.0.7-54-gac50ffc
Homebrew/homebrew-core (git revision 5162d; last commit 2016-10-14)
==> Homebrew Executable Path:
/usr/local/bin/brew
==> Homebrew Cellar Path:
/usr/local/Cellar
==> Homebrew Repository Path:
/usr/local/Homebrew
==> Homebrew Origin:
https://github.com/Homebrew/brew.git
==> Homebrew-Cask Version:
Homebrew-Cask 1.0.7-54-gac50ffc
caskroom/homebrew-cask (git revision e9b05; last commit 2016-10-14)
==> Homebrew-Cask Install Location:
<NONE>
==> Homebrew-Cask Staging Location:
/opt/homebrew-cask/Caskroom
==> Homebrew-Cask Cached Downloads:
/Users/alex/Library/Caches/Homebrew/Cask
8 files, 319.5M (warning: run "brew cask cleanup")
==> Homebrew-Cask Default Tap Path:
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask
==> Homebrew-Cask Alternate Cask Taps:
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-versions
==> Homebrew-Cask Default Tap Cask Count:
3382
==> Contents of $LOAD_PATH:
/usr/local/Homebrew/Library/Homebrew/cask/lib
/usr/local/Homebrew/Library/Homebrew
/Library/Ruby/Site/2.0.0
/Library/Ruby/Site/2.0.0/x86_64-darwin16
/Library/Ruby/Site/2.0.0/universal-darwin16
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/x86_64-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/universal-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/x86_64-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/universal-darwin16
==> Contents of $RUBYLIB Environment Variable:
<NONE>
==> Contents of $RUBYOPT Environment Variable:
<NONE>
==> Contents of $RUBYPATH Environment Variable:
<NONE>
==> Contents of $RBENV_VERSION Environment Variable:
<NONE>
==> Contents of $CHRUBY_VERSION Environment Variable:
<NONE>
==> Contents of $GEM_HOME Environment Variable:
<NONE>
==> Contents of $GEM_PATH Environment Variable:
<NONE>
==> Contents of $BUNDLE_PATH Environment Variable:
<NONE>
==> Contents of $PATH Environment Variable:
PATH="/Users/alex/anaconda/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/git/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Library/TeX/texbin:/usr/local/pspdev/bin:/usr/local/Homebrew/Library/Homebrew/cask/cmd:/usr/local/Homebrew/Library/Homebrew/shims/scm"
==> Contents of $SHELL Environment Variable:
SHELL="/bin/zsh"
==> Contents of Locale Environment Variables:
LANG="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
==> Running As Privileged User:
No
```
</details>

**Comment by miccal** [comment]
#25611

**Comment by jawshooah** [comment]
Closed in favor of Homebrew/brew#1315.
